ci/do_ci.sh

```diff
@@ -42,6 +42,12 @@ export BAZEL_EXTRA_TEST_OPTIONS="--test_env=ENVOY_IP_TEST_VERSIONS=v4only --test
 export ENVOY_CONTRIB_BUILD_TARGET="//source/exe:envoy-static"
 export ENVOY_CONTRIB_BUILD_DEBUG_INFORMATION="//source/exe:envoy-static.dwp"
 
+BAZEL_BUILD_EXTRA_OPTIONS="${BAZEL_BUILD_EXTRA_OPTIONS} --remote_cache=${BAZEL_REMOTE_CACHE}"
```

Comment:
are we certain this isn't overwriting something coming in from the yaml or something? Can we `+=`?
```suggestion
BAZEL_BUILD_EXTRA_OPTIONS+=" ${BAZEL_BUILD_EXTRA_OPTIONS} --remote_cache=${BAZEL_REMOTE_CACHE}"
```

Reply:
I had applied this suggestion, but I ended up changing it slightly, to
```diff
- BAZEL_BUILD_EXTRA_OPTIONS+=" ${BAZEL_BUILD_EXTRA_OPTIONS} --remote_cache=${BAZEL_REMOTE_CACHE}"
+ BAZEL_BUILD_EXTRA_OPTIONS+=" --remote_cache=${BAZEL_REMOTE_CACHE}"
```

In either case, we should respect any prior value of `BAZEL_BUILD_EXTRA_OPTIONS`. I expect this value to not be defined, as envoy-gloo does not specify this value in `ci/cloudbuild.yaml` (compare [envoy-gloo](https://github.com/solo-io/envoy-gloo/blob/main/ci/cloudbuild.yaml) with envoy-gloo-ee [ci/cloudbuild.yaml](https://github.com/solo-io/envoy-gloo-ee/blob/main/ci/cloudbuild.yaml#L12) and [top-level cloudbuild.yaml](https://github.com/solo-io/envoy-gloo-ee/blob/main/cloudbuild.yaml#L4))